hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/HoodieWriterUtils.scala

```diff
@@ -216,6 +217,34 @@ object HoodieWriterUtils {
     validateTableConfig(spark, params, tableConfig, false)
   }
 
+  /**
+   * This function adds specific rules to choose config key in table config for a given writer key.
+   *
+   * RULE 1: When
+   *   1. table version is 9,
+   *   2. writer key is a payload class key, and
+   *   3. table config has legacy payload class configured,
+   * then
+   *   return legacy payload class key.
+   *
+   * Basic rule:
+   *   return writer key.
+   */
+  def getKeyInTableConfig(key: String, tableConfig: HoodieConfig): String = {
```

Comment:
can we rename, looks like the method only handle payload class related logic.

Reply:
renamed this to `getPayloadClassConfigKeyFromTableConfig` .